test/plug_test.exs

```diff
@@ -103,6 +103,17 @@ defmodule PlugTest do
     assert load_resource(conn, opts) == expected
   end
 
+  test "it loads the resource correctly with opts[:id] specified" do
+    opts = [model: Post, id: "user_id"]
```

Comment:
Do you wanna call it `id_name` instead of just `id` to go with `resource_name` and not confuse with the actual `id`?